File: Clase0606/persistencia/Persona.java
package Clase0606.persistencia;
import java.io.*;

public class Persona {
    private String rutaBase = "F:/program/aux_121/Programacion-2-I2025/Clase0606/personaJava/";

    private int id;
    private String nom;
    private int edad;
    private String archivo;

    public Persona(int id, String nom, int edad) {
        this.id = id;
        this.nom = nom;
        this.edad = edad;
        this.archivo = rutaBase+"persona" + id + ".txt";
    }

    public Persona(String rutaArchivo) throws IOException {
        this.archivo = rutaArchivo;

        File archivo = new File(this.archivo);

        if (!archivo.exists()) {
            throw new FileNotFoundException("El archivo no existe: persona" +this.id+".txt" );
        }else{
            BufferedReader br = new BufferedReader(new FileReader(archivo));

            this.id = Integer.parseInt(br.readLine());
            this.nom = br.readLine();
            this.edad = Integer.parseInt(br.readLine());
            
            br.close();
        }
    }

    public void guardar() throws IOException {
        File carpeta = new File(rutaBase);
        
        if (!carpeta.exists()) {
            carpeta.mkdirs(); // Crea la carpeta si no existe
        }

        File archivo = new File(this.archivo);
        if(!archivo.exists()) {
            archivo.createNewFile(); // Crea el archivo si no existe
        }

        BufferedWriter bw = new BufferedWriter(new FileWriter(this.archivo));
        bw.write(this.id + "\n");
        bw.write(this.nom + "\n");
        bw.write(this.edad + "\n");
        bw.close();
    }

    public void actualizar() throws IOException {
        eliminar();
        guardar();
    }

    
    public void eliminar() {
        File f = new File(this.archivo);
        if (f.exists()) f.delete();
    }

    @Override
    public String toString() {
        return "ID: " + id + ", Nombre: " + nom + ", Edad: " + edad;
    }

    public String getNom(){
        return this.nom;
    }
}

File: Clase0606/Persona.py
import os
import json

rutaBase = "F:/program/aux_121/Programacion-2-I2025/Clase0606/personaPython"
class Persona:
    def __init__(self, id=None, nom=None, edad=None):
        self.id = id
        self.nom = nom
        self.edad = edad
        self.archivo = f"{rutaBase}/persona{self.id}.json"

    def cargarDatos(self, rutaArchivo):
        self.archivo = rutaArchivo
        if os.path.isfile(self.archivo):
            with open(self.archivo, 'r') as archivo:
                datos = json.load(archivo)
                self.id = datos.get("id")
                self.nom = datos.get("nom")
                self.edad = datos.get("edad")
                print(f"Datos cargados: persona{self.id}.json")
        else:
            print(f"El archivo persona{self.id}.json no existe.")
        
    def guardarDatos(self):
        os.makedirs(rutaBase, exist_ok=True)
        
        if not os.path.exists(self.archivo):
            with open(self.archivo, 'w') as archivo:
                datos = {
                    "id": self.id,
                    "nom": self.nom,
                    "edad": self.edad
                }
                
                with open(self.archivo, 'w') as archivo:
                    json.dump(datos, archivo, indent=4)
            print(f"Datos guardados en: persona{self.id}.json")

        
    def actualizarDatos(self):
        with open(self.archivo, 'r') as archivo:
            datos = json.load(archivo)
            datos["nom"] = self.nom
            datos["edad"] = self.edad
            with open(self.archivo, 'w') as archivo:
                json.dump(datos, archivo, indent=4)

    def eliminarDatos(self):
        if os.path.isfile(self.archivo):
            os.remove(self.archivo)
            print(f"Archivo persona{self.id}.json eliminado.")
        else:
            print(f"El archivo persona{self.id}.json no existe.")
    
    def __str__(self):
        return f"ID: {self.id}, Nombre: {self.nom}, Edad: {self.edad}"
    
